### PR TITLE
Rados Bugzilla automations and TFA fixes

### PR DESCRIPTION
**Pull request by pdhiran**
1. Verification that mon daemons cannot be added without location attributes in stretch mode
2. Verification of OSD behaviour upon marking OSD IN and OUT of the cluster

### CODE REVIEW

**Comment by pdhiran** [comment]
Pass logs:
5.x : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-RUEY87

File: tests/rados/pool_tests.py [hunk 1]
```diff
@@ -1131,6 +1143,117 @@ def run(ceph_cluster, **kw):
...
+                != selected_osd_details_init["crush_weight"]
+            ):
+                log.error(
+                    f"Crush weight modified post marking the OSD.{selected_osd} out."
+                    f"Before : {selected_osd_details_init['crush_weight']} "
+                    f"After : {selected_osd_details_post['crush_weight']}"
+                )
+                raise Exception("OSD Crush weights modified error")
+
+            if (
```
Comment (pdhiran) [inline tests/rados/pool_tests.py:1194]: reweight should be 0 once OSD is marked out.

File: tests/rados/pool_tests.py [hunk 2]
```diff
@@ -1131,6 +1143,117 @@ def run(ceph_cluster, **kw):
...
+                log.error(f"OSD.{selected_osd} not marked out error")
+                raise Exception("OSD not marked out error")
+            log.info(f"OSD: {selected_osd} marked out successfully. ")
+
+            osd_tree = rados_obj.run_ceph_command(cmd="ceph osd tree")
+            for entry in osd_tree["nodes"]:
+                if entry["id"] == selected_osd:
+                    log.info(f"Selected OSD details After marking out : {entry}")
+                    selected_osd_details_post = entry
+                    break
```
Comment (harshkumarRH) [inline tests/rados/pool_tests.py:1181]: Just a suggestion, we can skip this code block by utilizing exiting method for `ceph osd df` and use it to fetch the stat for the concerned osd itself
```
[root@ceph-hakumar-mm6odk-node7 ~]# ceph osd df osd.0
ID  CLASS  WEIGHT   REWEIGHT  SIZE    RAW USE  DATA    OMAP  META    AVAIL   %USE  VAR   PGS  STATUS
 0    hdd  0.02440   1.00000  25 GiB   46 MiB  10 MiB   0 B  36 MiB  25 GiB  0.18  1.00   10      up
                       TOTAL  25 GiB   46 MiB  10 MiB   0 B  36 MiB  25 GiB  0.18                   
MIN/MAX VAR: 1.00/1.00  STDDEV: 0
[root@ceph-hakumar-mm6odk-node7 ~]# ceph osd df osd.0 -f json-pretty

{
    "nodes": [
        {
            "id": 0,
            "device_class": "hdd",
            "name": "osd.0",
            "type": "osd",
            "type_id": 0,
            "crush_weight": 0.0243988037109375,
            "depth": 2,
            "pool_weights": {},
            "reweight": 1,
            "kb": 26210304,
            "kb_used": 46600,
            "kb_used_data": 10240,
            "kb_used_omap": 0,
            "kb_used_meta": 36352,
            "kb_avail": 26163704,
            "utilization": 0.17779267268323176,
            "var": 1,
            "pgs": 10,
            "status": "up"
        }
    ],
    "stray": [],
    "summary": {
        "total_kb": 26210304,
        "total_kb_used": 46600,
        "total_kb_used_data": 10240,
        "total_kb_used_omap": 0,
        "total_kb_used_meta": 36352,
        "total_kb_avail": 26163704,
        "average_utilization": 0.17779267268323176,
        "min_var": 1,
        "max_var": 1,
        "dev": 0
    }
}```

Comment (pdhiran) [reply tests/rados/pool_tests.py:1181]: I'll have a backlog task to create a new method for this use-case, something like `def get_osd_details` to fetch all the details for a particular OSD.

File: tests/rados/pool_tests.py (hunk 1 above)
Comment (pdhiran) [reply tests/rados/pool_tests.py]: Modified the code.

File: tests/rados/pool_tests.py (hunk 2 above)
Comment (pdhiran) [reply tests/rados/pool_tests.py:1181]: Backlog task : https://issues.redhat.com/browse/RHCEPHQE-15034

File: ceph/rados/core_workflows.py [hunk 3]
```diff
@@ -1545,6 +1545,57 @@ def set_mclock_parameter(
...
+    def update_osd_state_on_cluster(self, osd_id, state):
+        """
+        Method to update the state of the OSD in the cluster. This method can mark the osd "in, out, up, down"
+        Args:
+            osd_id: ID of the osd to be operated
+            state: Property that needs to be updated.
+                  Allowed states: "in, out, up, down"
+        Returns: True -> pass, false -> fail
+        """
+        allowed_states = {"in", "out", "up", "down"}
```
Comment (harshkumarRH) [inline ceph/rados/core_workflows.py:1557]: Is there a reason, why this is not a list?

**Review by neha-gangadhar — COMMENTED** [review]
lgtm

File: tests/rados/pool_tests.py [hunk 4]
```diff
@@ -1131,6 +1143,114 @@ def run(ceph_cluster, **kw):
...
+    if config.get("Verify_osd_in_out_behaviour"):
+        log.info(
+            "Test to verify the behaviour of the OSD when it is marked out and back in to the cluster"
+        )
+        try:
+            osd_tree = rados_obj.run_ceph_command(cmd="ceph osd tree")
+            osd_list = []
+            for entry in osd_tree["nodes"]:
+                if entry["id"] > 0:
+                    osd_list.append(entry["id"])
```
Comment (harshkumarRH) [inline tests/rados/pool_tests.py:1155]: What exactly are we trying to achieve here?

File: ceph/rados/core_workflows.py (hunk 3 above)
Comment (pdhiran) [reply ceph/rados/core_workflows.py:1557]: That's the max amount of allowed states, and no more would be added or removed.

Did not see any reason to make it a mutable element.

File: tests/rados/pool_tests.py (hunk 4 above)
Comment (pdhiran) [reply tests/rados/pool_tests.py:1155]: A crude way to fetch all the OSDs on the cluster.

**Review by harshkumarRH — APPROVED** [review]
Looks good

File: tests/rados/pool_tests.py (hunk 4 above)
Comment (pdhiran) [reply tests/rados/pool_tests.py:1155]: Use method : def get_active_osd_list(self) here once the PR : https://github.com/red-hat-storage/cephci/pull/3806/ is merged.

**Review by neha-gangadhar — APPROVED** [review]
Approved

**Pull request merged**